### ISSUE TITLE
Compilation error in executor_supervisor.ex

### ISSUE THREAD

**Issue by sublimecoder** [quantum-elixir/quantum-core#370]
I'm getting the following error in 2.3.2 but not in 2.2.7 any idea why this is happening?

```bash
== Compilation error in file lib/quantum/executor_supervisor.ex ==
** (CompileError) lib/quantum/executor_supervisor.ex:53: unknown key :execution_broadcaster for struct Quantum.ExecutorSupervisor.InitOpts
```

**Comment by kerrd89** [comment]
I am seeing the same.  More specifically, I can compile w/ 2.3.1 (labeled retired on hex) but not with 2.3.2, but 2.3.1 does not appear to have updated features from 2.3.

elixir version: 1.6.4

**Comment by maennchen** [comment]
Did you check if the problem persists when you rebuild completely? (Just throw away the `_build` and `deps` folder and reinstall the deps)

**Comment by kerrd89** [comment]
Yes, even after removing the `_build` and `deps` folder, the dependency fails to compile with the same warning.

**Comment by kerrd89** [comment]
In `lib/quantum/executor_supervisor.ex:53` we need to update the key to `:execution_broadcaster_reference` for the if -> true case, same as the else clause beneath.

```
  if Util.gen_stage_v12?() do
    def init(
          %InitOpts{
            execution_broadcaster: execution_broadcaster
          } = opts
        ) do
```

**Comment by maennchen** [comment]
@kerrd89 I see. I'm going to create a PR in a few minutes.

**Comment by maennchen** [comment]
@kerrd89 Fix in #372

**Comment by jbavari** [comment]
This is working now! Let's close this bad boy.

**Comment by kerrd89** [comment]
Thanks for the quick turnaround!

**Comment by maennchen** [comment]
I'll try to get the other PR running as well and will then create a new release. (#369)

**Comment by maennchen** [comment]
Will be released with #373
